app/components/item-thumbnail-view.js

```diff
@@ -0,0 +1,45 @@
+import { bind } from "@ember/runloop";
+import Component from "@ember/component";
+import { inject as service } from "@ember/service";
+import cloudinaryImage from "../mixins/cloudinary_image";
+
+export default Component.extend(cloudinaryImage, {
+  loading: true,
+  store: service(),
+
+  changedSrc: function() {
+    if (!this.is_cached(this.get("src"))) {
+      this.set("loading", true);
+    }
+  }.observes("src"),
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    this.set("cloudinaryId", this.get("cloudinaryId"));
+    this.set("src", this.generateUrl(500, 500, true));
+    const pkgTypeType = this.get("store").peekRecord(
+      "package_type",
+      this.get("item").attributes.package_type_id
+    );
+    this.set("pkgTypeType", pkgTypeType);
+  },
```

Comment:
```javascript
  // no need for didReceiveAttrs, use computed properties
  src: Ember.computed('cloudinaryId', function () {
     return this.generateUrl(500, 500, true);
   }),
  
  pkgType: Ember.computed('packageTypeId', function () {
    // - I renamed  pkgTypeType into pkgType (didn't make any sense)
    // - you can set the packageTypeId one the the component instead of setting the item
    //  otherwise this component is not generic since it won't work with normal models (since you do `.attributes`)
    return this.get("store").peekRecord(
      "package_type",
      this.get("packageTypeId")
    );
  }),
```